ipatests/test_integration/test_ipa_ipa_migration.py

```diff
@@ -920,3 +923,322 @@ def test_ipa_migrate_stage_mode_with_invalid_cert(self):
+        result2 = self.replicas[0].run_command(['id', username])
+        assert result1.stdout_text == result2.stdout_text
+
+    def test_ipa_migrate_subids(self, empty_log_file):
+        """
+        This testcase checks that subids for users are migrated
+        to the local server from the remote server
+        """
+        user_name = 'admin'
+        CMD_MSG = "1 subordinate id matched"
+        run_migrate(
+            self.replicas[0],
+            "prod-mode",
+            self.master.hostname,
+            "cn=Directory Manager",
+            self.master.config.admin_password,
+            extra_args=['-n']
+        )
+        cmd = self.replicas[0].run_command(
+            ['ipa', 'subid-find',
+             '--owner', user_name]
+        )
+        assert cmd.returncode == 0
+        assert CMD_MSG in cmd.stdout_text
+
+    def test_ipa_migrate_check_service_status(self):
+        """
+        This testscase checks that ipactl and sssd
+        services are running post ipa-migrate tool
+        successful runs completed
+        """
+        run_migrate(
+            self.replicas[0],
+            "prod-mode",
+            self.master.hostname,
+            "cn=Directory Manager",
+            self.master.config.admin_password,
+            extra_args=['-n']
+        )
+        cmd1 = self.replicas[0].run_command([
+            "ipactl", "status"
+        ])
+        assert cmd1.returncode == 0
+        cmd2 = self.replicas[0].run_command([
+            "systemctl", "status", "sssd"
+        ])
+        assert cmd2.returncode == 0
+
+    def test_custom_idrange_is_migrated(self, empty_log_file):
+        """
+        This testcase checks that custom idrange is migrated
+        from remote server to local server in production
+        mode.
+        """
+        range_name = "testrange"
+        CMD_OUTPUT = (
+            "---------------\n"
+            "1 range matched\n"
+            "---------------\n"
+            "  Range name: testrange\n"
+            "  First Posix ID of the range: 10000\n"
+            "  Number of IDs in the range: 10000\n"
+            "  First RID of the corresponding RID range: 300000\n"
+            "  First RID of the secondary RID range: 400000\n"
+            "  Range type: local domain range\n"
+            "----------------------------\n"
+            "Number of entries returned 1\n"
+            "----------------------------\n"
+        )
+        run_migrate(
+            self.replicas[0],
+            "prod-mode",
+            self.master.hostname,
+            "cn=Directory Manager",
+            self.master.config.admin_password,
+            extra_args=['-n']
+        )
+        cmd = self.replicas[0].run_command(
+            ["ipa", "idrange-find", range_name])
+        assert CMD_OUTPUT in cmd.stdout_text
+
+    def test_automountlocation_is_migrated(self):
+        """
+        This testcase checks that automount location/maps
+        and keys are migrated.
+        """
```

Comment:
This test is failing because of the way the tests are executed. The source server is prepared during the installation, and ipa-migrate is ran multiple times. The entries that were created during preparation are migrated on the first call to ipa-migrate, not on later calls.
It would be a better strategy to add the entries relevant to the test in a fixture for the test, and always call empty_log_file.